ci/check-tidy.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+set -e
```

Comment:
What about adding this as a target `tidy`, and then adding that to `lint`? Would make contributing a bit faster if linting can fail locally.

Reply:
I wouldn't recommend doing that, as I have observed that sometimes the `go.mod` and `go.sum` files generated by `go mod tidy` sometimes depend on the Go minor version used (maybe because of issues early on in the Go module implementation?). And the Go version used by someone locally may not match the Go version used for CI.